library/src/main/java/dev/hichamboushaba/suspendactivityresult/ActivityProvider.kt

```diff
@@ -21,7 +21,7 @@ internal object ActivityProvider : Application.ActivityLifecycleCallbacks {
             val activity = it.get()
             activity != null && activity.lifecycle.currentState.isAtLeast(Lifecycle.State.INITIALIZED)
         }
-        .map { it.get()!! }
+        .map { it.get() }
```

Comment:
I think then better to add .filterNotNull() to _activityFlow, because if it.get() return null then will be runtimeExeption. It will be bad. My variant is that:

```kotlin
val activityFlow: Flow<ComponentActivity> =
        _activityFlow
            .asStateFlow()
            .distinctUntilChanged { old, new -> old.get() === new.get() }
            .map { it.get() }
            .filterNotNull()
            .filter { it.lifecycle.currentState.isAtLeast(Lifecycle.State.INITIALIZED) }
```

Reply:
And applying 'distinctUntilChanged' to StateFlow has no effect. Because StateFlow already behaves as if distinctUtilChanged operator is applied

Reply:
> And applying 'distinctUntilChanged' to StateFlow has no effect. Because StateFlow already behaves as if distinctUtilChanged operator is applied

`StateFlow` will use the default `equals` implementation for comparison, that's not what we are doing, we are applying the `distinctUtilChanged` on the `Activity` instance, since we may send the same activity multiple times, in different `WeakReference` instances.

> I think then better to add .filterNotNull() to _activityFlow, because if it.get() return null it will be NPE. It can be bad and dangerous. My variant is that:

This is the same logic as I'm doing, I'm already filtering non-null activity instances on the `filter` block, so `it.get()` won't throw any NPE.

Reply:
> StateFlow will use the default equals implementation for comparison, that's not what we are doing, we are applying the distinctUtilChanged on the Activity instance, since we may send the same activity multiple times, in different WeakReference instances.
It's important information, then could be to add comment for it case. I fixed:
https://github.com/hichamboushaba/SuspendActivityResult/pull/1/commits/60991fde23e1876eb11a56ce128e71c04b14b9b0

Reply:
> This is the same logic as I'm doing, I'm already filtering non-null activity instances on the filter block, so it.get() won't throw any NPE.

I do not agree with you.
1. WeakReference can be null anytime and it can be happen in moment then will executing .map { it.get()!! } and then will be NPE. 
2. In first initialization there is a chance can be NPE, because MutableStateFlow emit null: 
> private val _activityFlow = MutableStateFlow(WeakReference<ComponentActivity>(null))

And finally, why ide highlights in place for insert "!!", because kotlin null safety and do it that it do kotlin contract and filtering non-null activity instances on the filter block is not checking on kotlin-contract.

If that reasons don't look convincing to you, i fix it, but i will be sad.

Reply:
> In first initialization there is a chance can be NPE, because MutableStateFlow emit null:

As I said before, it won't cause any NPE, I had a check on null values on the `filter` block before 🙂.

I'm merging this as is, since it's the same logic as before, and won't hurt.
But just for future contributions, please limit changes to the relevant parts of the PR 🙏 , for example for this, it should've been just the new extensions, if you want to change something else, we can discuss it in a separate PR 🙂.